snuba/utils/health_info.py

```diff
@@ -0,0 +1,181 @@
+from __future__ import annotations
+
+import itertools
+import logging
+import os
+import time
+from collections import defaultdict
+from typing import (
+    Any,
+    Dict,
+    List,
+    Mapping,
+    MutableMapping,
+    Optional,
+    Set,
+    Tuple,
+    Union,
+    cast,
+)
+
+import simplejson as json
+
+from snuba import environment, settings
+from snuba.clickhouse.errors import ClickhouseError
+from snuba.clusters.cluster import (
+    ClickhouseClientSettings,
+    ConnectionId,
+    UndefinedClickhouseCluster,
+)
+from snuba.datasets.factory import get_dataset, get_enabled_dataset_names
+from snuba.datasets.schemas.tables import TableSchema
+from snuba.datasets.storage import Storage
+from snuba.environment import setup_logging
+from snuba.utils.metrics.wrapper import MetricsWrapper
+
+metrics = MetricsWrapper(environment.metrics, "api")
+setup_logging(None)
+logger = logging.getLogger("snuba.health")
+
+
+def shutdown_time() -> Optional[float]:
+    try:
+        return os.stat("/tmp/snuba.down").st_mtime
+    except OSError:
+        return None
+
+
+try:
+    IS_SHUTTING_DOWN = False
+    import uwsgi
+except ImportError:
+
+    def check_down_file_exists() -> bool:
+        return False
+
+else:
+
+    def check_down_file_exists() -> bool:
+        global IS_SHUTTING_DOWN
+        try:
+            m_time = shutdown_time()
+            if m_time is None:
+                return False
+
+            start_time: float = uwsgi.started_on
+            IS_SHUTTING_DOWN = m_time > start_time
+            return IS_SHUTTING_DOWN
+        except OSError:
+            return False
+
+
+def get_health_info(thorough: Union[bool, str]) -> Tuple[str, int, Dict[str, str]]:
```

Comment:
nit: this tuple type doesn't tells me little about what the function returns. a `dataclass` with named fields would be a better experience for me as a reader/user of the code